compiler/src/dmd/expressionsem.d

```diff
@@ -13174,10 +13174,6 @@ bool checkSharedAccess(Expression e, Scope* sc, bool returnRef = false)
         {
             if (e.thisexp)
                 check(e.thisexp, false);
-            // Note: This handles things like `new shared(Throwable).msg`,
-            // where accessing `msg` would violate `shared`.
-            if (e.newtype.isShared())
-                return sharedError(original);
```

Comment:
This should be checked for in DotVarExp, not here since it essentially forbids creating any shared instances via new.